src/Microsoft.AspNetCore.Authentication/PolicySchemeHandler.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    /// <summary>
+    /// PolicySchemes are used to redirect authentication methods to another scheme.
+    /// </summary>
+    public class PolicySchemeHandler : SignInAuthenticationHandler<PolicySchemeOptions>
+    {
+        public PolicySchemeHandler(IOptionsMonitor<PolicySchemeOptions> options, ILoggerFactory logger, UrlEncoder encoder, ISystemClock clock) : base(options, logger, encoder, clock)
+        { }
+
+        protected override Task HandleChallengeAsync(AuthenticationProperties properties)
```

Comment:
This works because the base class handles all of the forwarding so this only gets invoked if there is no forwarding? At which point you fall back to the system default?  Which may be this handler? 

It seems like it would be a lot safer to require forwarding for this handler and throw if it ever fell through to the local implementation.

Reply:
Yeah so previous iteration of this required an 'anchor' basically a default forward.  The reason I'm not sure that's what we want anymore, is this should also be the base class of choice for plugging in custom 'policies' that are just gluing other schemes together, like if you want to have a "AllBearer" scheme that implements authentication by merging Bearer1 + Bearer2 together.  Requiring something like that to forward seems wrong.

Reply:
Why would they extend this class vs SignInAuthenticationHandler

Reply:
Because they don't might not want to implement SignIn?  I mean that works too, but this one isn't abstract, so they don't have to implement anything

Reply:
Make it two classes? What for the build in forwarders that requires a default, and one for customization that doesn't?

Reply:
Yes, throwing NotImplemented is my main suggestion.

Reply:
Ok that's fine, derived policies can always add fallback to the default schemes if they want anyways... updated